src/fu-keyring.c

```diff
@@ -198,222 +70,63 @@ fu_keyring_add_public_keys (FuKeyring *keyring, const gchar *dirname, GError **e
 		if (filename == NULL)
 			break;
 		path_tmp = g_build_filename (dirname, filename, NULL);
-		if (!fu_keyring_add_public_key (keyring, path_tmp, error))
-			return FALSE;
+		for (guint i = 0; i < priv->cryptos->len; i++) {
+			FuCrypto *crypto = g_ptr_array_index (priv->cryptos, i);
+			g_debug ("adding public key %s to crypto provider %s",
+				 path_tmp, fu_crypto_get_name (crypto));
+			if (!fu_crypto_add_public_key (crypto, path_tmp, error))
```

Comment:
`fu_crypto_add_public_key` returns `TRUE` if it can't handle a given type of public key, which means we don't have a way to tell that we have no provider enabled that can handle a given key. If this was desired `fu_crypto_add_public_key` could maybe return an UNSUPPORTED error that we match and if we only get those back we fail?